packages/editor/src/lib/editor/types/SvgExportContext.tsx

```diff
@@ -29,10 +30,29 @@ export interface SvgExportContext {
 	 */
 	waitUntil(promise: Promise<void>): void
 
+	/**
+	 * Resolve an asset URL in the context of this export. Supply the asset ID and the width in
+	 * shape-pixels it'll be displayed at, and this will resolve the asset according to the export
+	 * options.
+	 */
+	resolveAssetUrl(assetId: TLAssetId, width: number): Promise<string | null>
+
 	/**
 	 * Whether the export should be in dark mode.
 	 */
 	readonly isDarkMode: boolean
+
+	/**
+	 * The scale of the export - how much CSS pixels will be scaled up/down by.
+	 */
+	readonly scale: number
+
+	/**
+	 * If the ultimate target for this export is a raster image, this is the number of raster pixels
+	 * in the resulting bitmap per CSS pixel in the resulting SVG. If this value is null, the export
```

Comment:
nit, seems hard to read: "If this value is null or the export is to SVG, then the asset won't be downscaled."

Reply:
This is a bit tricky to write for as in the context where this is used, it's something we give the user rather than the user giving to us. So they should be the ones applying the downscaling, if they're writing a custom shape where that's relevant. I've updated this to try and make it make more sense though.